packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -201,17 +215,27 @@ const HoverCardContentImpl = React.forwardRef<
 
   return (
     <PortalWrapper>
-      <PopperPrimitive.Content
-        {...popperScope}
-        {...contentProps}
-        ref={composedRefs}
-        style={{
-          ...contentProps.style,
-          // re-namespace exposed content custom property
-          ['--radix-hover-card-content-transform-origin' as any]:
-            'var(--radix-popper-transform-origin)',
-        }}
-      />
+      <DismissableLayer
+        asChild
+        disableOutsidePointerEvents={disableOutsidePointerEvents}
```

Comment:
I don't think all props should be exposed. It should be equivalent to those exposed by non-modal primitives I believe.

Reply:
Looks like its only `disableOutsidePointerEvents` that I needed to remove. Let me know if I've misunderstood the non-modal code though.

Reply:
I think `onDismiss` also is never exposed.

Reply:
I had already omitted that one. I've switched it now though to manually expose the relevant ones instead of omitting the unavailable ones. It doesn't make sense to `extend` a type if the props aren't spread on that component.